### PR TITLE
fix(challenges): Improve local scope challenge tests

### LINKED ISSUE #13409: Local Scope and Functions issue with commented code.
Challenge [Local Scope and Functions](https://www.freecodecamp.com/challenges/local-scope-and-functions#?solution=%0Afunction%20myLocalScope(myVar)%20%7B%0A%20%20myVar%20%3D%205%3B%20%0A%20%20%0A%20%20console.log(myVar)%3B%0A%7D%20%0A%2F%2Fvar%20myVar%20%3D%20300%3B%20%20%0A%2F%2F%20line%207%20is%20what%20I%20am%20%20talking%20about..%20If%20I%20commend%20it%2C%20I%20succesful%20finish%20the%20challenge..%20but%20when%20I%20delete%20line%207%2C%20so%20it%20doesn%27t%20work...%20and%20if%20I%20UnComment%20it%2C%20so%20it%20doesn%27t%20work%20too..%20That%27s%20a%20bug%0AmyLocalScope()%3B%0A%0A%2F%2F%20Run%20and%20check%20the%20console%0A%2F%2F%20myVar%20is%20not%20defined%20outside%20of%20myLocalScope%0A%0A%2F%2F%20Now%20remove%20the%20console%20log%20line%20to%20pass%20the%20test%0A%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:50.0) Gecko/20100101 Firefox/50.0</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

My code:
```javascript

function myLocalScope(myVar) {
  myVar = 5; 
  
  console.log(myVar);
} 
//var myVar = 300;  
// LINE 7 IS WHAT I AM TALKING ABOUT.. IF I COMMENT  IT, I SUCCESFUL FINISH THE CHALLENGE.. BUT IF I DELETE LINE 7, IT DOESN'T WORK... and if I UnComment it, so it doesn't work too.. That's a bug
myLocalScope();

// Run and check the console
// myVar is not defined outside of myLocalScope

// Now remove the console log line to pass the test


```

### PR DESCRIPTION
**Pull request by systimotic**
#### Pre-Submission Checklist
- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change
- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:
- [x] Tested changes locally.
- [x] Closes currently open issue (replace XXXX with an issue no): Closes #13409

#### Description
This PR updates the challenge as discussed in the issue. See regexr.com/3gk6t for the regex with some test cases.

### CODE REVIEW

**Comment by systimotic** [comment]
Regarding that first camperbot message: we should probably get that updated, it conflicts with the commitzen guidelines

**Comment by QuincyLarson** [comment]
@systimotic Thanks for your patience. I just QA'd this. 

This does indeed work, though I was a bit confused as to which console.log line to update. I also am not a fan of the "follow the directions in the code" part. I feel like we need to figure out a more elegant way of conveying this.

File: seed/challenges/02-javascript-algorithms-and-data-structures/basic-javascript.json [hunk 1]
```diff
@@ -2522,11 +2522,11 @@
         "(function() { return logOutput || \"console.log never called\"; })();"
       ],
       "solutions": [
-        "function myLocalScope() {\n  'use strict';\n  \n  var myVar;\n  console.log(myVar);\n}\nmyLocalScope();"
+        "function myLocalScope() {\n  'use strict';\n  \n  var myVar = 100;\n  console.log(myVar);\n}\nmyLocalScope();"
       ],
       "tests": [
         "assert(typeof myVar === 'undefined', 'message: No global <code>myVar</code> variable');",
-        "assert(/var\\s+myVar/.test(code), 'message: Add a local <code>myVar</code> variable');"
+        "assert(/^[^\\/\\n]*var[^\\S\\n]+myVar[^\\S\\n]*=[^\\S\\n]*100[^\\S\\n]*;?.*$/m.test(code), 'message: Add a local <code>myVar</code> variable with a value of <code>100</code>');"
```
Comment (QuincyLarson) [inline seed/challenges/02-javascript-algorithms-and-data-structures/basic-javascript.json]: This does indeed work, though I was a bit confused as to which console.log line to update. I also am not a fan of the "follow the directions in the code" part. I feel like we need to figure out a more elegant way of conveying this.

**Review by QuincyLarson — CHANGES_REQUESTED** [review]

**Comment by raisedadead** [comment]
@systimotic ping!

**Comment by raisedadead** [comment]
Closing decayed, please re-open if you see this and have additional clarification as requested.

**Pull request closed without merging**